### ISSUE TITLE
Allow multiple release events for an album

### ISSUE THREAD

**Issue by riipah** [VocaDB/vocadb#132]

**Comment by romanrm** [comment]
Just checked to see if this has been suggested already.
E.g. https://vocadb.net/S/219995 would qualify both for the real-life event of Comiket 95, and the virtual Kagamine Birthday 2018 event.
Seems the current practice is to unlink such songs from virtual events as the real one takes precedence, but I guess that's not ideal, now you can't find it anymore from the anniversary event.

**Comment by riipah** [comment]
You're correct that for physical + virtual events it makes more sense. I still don't like the idea of marking all physical events. It's always more complicated when there is multiple instead of one. Both UI and performance. But it's possible, I just have more urgent things to do first, so this will take some time.

**Comment by andreoda** [comment]
I still think this feature would be nice to have. That way we have "cleaner" event pages.

**Comment by ycanardeau** [comment]
> Most useful for character anniversaries.

@andreoda Could you please write down some examples?

**Comment by andreoda** [comment]
@ycanardeau 

Let's say we have a song called "xxx" by Hatsune Miku.
That song was released for the Song Contest 123, it was also released for Miku's birthday and was sold at Comiket 123. riipah says only the release event should be considered but imo it is way better to make it possible to add multiple events to songs or albums and leave the "release". I think the events category is not only for releases or it would make no sense to have birthday events. 
I was once told that events which have a date should be made into an event for the event category instead of a tag. It makes no sense, if there can not be multiple added. Ofc then I would prefer either making all as tags or making it possible to add multiple to a song or album.
I thought the database is for statistics, collecting information and such. If I use the search page and filter by a specific event, for example Miku's birthday 2019 and can not find specific songs there because they were added to another event, then the search is useless. It would only make sense to make all tags then, especially birthday events, but again, there is no way to add dates to tags and they will also not appear on the front page.
Btw, an example for this is this https://vocadb.net/E/2042/kagamine-birthday-2019 or this https://vocadb.net/E/3007/lukas-birthday-2021 entries are missing there because they can not be added. Same problem goes for contests or "festivals" that have a time range. Their belonged entries can not be added, so they are missing and the point of having those events is 0.

I hope I explained this well enough since I am super bad at arguments even tho I have a clear picture of what I mean in mind.

**Comment by riipah** [comment]
No one is denying that it would be useful, just that there are other competing objectives: 
* Features/fixes that are more urgent
* Making sure that it doesn't hurt performance
* Making sure that other features related to release events, such as publish date inference, still work properly

Also, consider the following situation: if the artist sells the album at Comiket 39, then again at Comiket 40, 41, 42. This is very common. Do you add all those events? Will it become convoluted if there's 10 events? What will be the guidance?

**Comment by andreoda** [comment]
I understand your points. Those are totally valid.

About the situation: I actually don't care much about release events or conventions where albums were sold. In a statistic and sorting perspective, I care more about events like contests and anniversaries or other celebrations.

Maybe those album selling conventions should get a whole new category but that would take a lot of work and time to implement. A whole new section in an album entry, for example under the "event" section. Something like "released at..." or just like your initial idea "release event". In that section only those selling/shop events like comiket etc would be mentioned. It is way clearer to sort it that way and people could check way better when can I buy that album or where and when was it actually sold.
That is a whole different story/idea tho.

Thank you for replying.

**Comment by riipah** [comment]
So one original release event, multiple distribution events, contents and anniversary. Perhaps the system could use the dates and event categories to determine how each one should be shown.

**Comment by riipah** [comment]
Some open issues to be discussed:

* Could someone do mockups how the user interface would look with multiple events?
* If we have many events (more than 5?), should the others be hidden, same as song lists and derivative versions?
* What are the headers? Should there still be header for original release (the first one?)
  * How do we determine if the song/album was actually released for that event? Comparing dates? Sometimes song/album is associated with birthday event years later.
* Should events always be shown ordered by date?
* Should the guidelines allow associating album with all distribution events (Comiket etc.) where it was sold and all events where it was mentioned (such as birthday, even if much later)?

**Comment by andreoda** [comment]
- If we have many events (more than 5?), should the others be hidden, same as song lists and derivative versions?
A: In song or album entries: I think they could be either listed vertically or horizontally like the tag system. Maybe like the tag system having the hidden feature and when clicking on it, it reveals more.
The release event should always stay open, not hidden and as a single field. The additional events should get an own field under the release event field.
